pkg/networkservice/mechanisms/memif/client.go

```diff
@@ -70,23 +71,33 @@ func (m *memifClient) Close(ctx context.Context, conn *networkservice.Connection
 	if err != nil {
 		return nil, err
 	}
-	m.appendInterfaceConfig(ctx, conn)
-	return rv, err
+	if err := m.appendInterfaceConfig(ctx, conn); err != nil {
+		return nil, err
+	}
+	return rv, nil
 }
 
-func (m *memifClient) appendInterfaceConfig(ctx context.Context, conn *networkservice.Connection) {
+func (m *memifClient) appendInterfaceConfig(ctx context.Context, conn *networkservice.Connection) error {
 	if mechanism := memif.ToMechanism(conn.GetMechanism()); mechanism != nil {
 		conf := vppagent.Config(ctx)
+		socketFileURL, err := url.Parse(mechanism.GetSocketFileURL())
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		if socketFileURL.Scheme != "file" {
+			return errors.Errorf("url scheme must be 'file' actual: %q", socketFileURL)
```

Comment:
Should we add a trace for this branch?

Reply:
errors.Errorf here should include the Stack as well :)